src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs

```diff
@@ -510,6 +510,7 @@ internal static bool PreventsSuccessfulDelegateConversion(ErrorCode code)
                 case ErrorCode.ERR_QueryRangeVariableSameAsTypeParam:
                 case ErrorCode.ERR_DeprecatedCollectionInitAddStr:
                 case ErrorCode.ERR_DeprecatedSymbolStr:
+                case ErrorCode.ERR_FeatureInPreview: // LangVersion should never change how we bind things
```

Comment:
>case ErrorCode.ERR_FeatureInPreview: [](http://example.com/codeflow?start=16&length=36)

Isn't there a chance to change existing behavior if we modify this list? The motivation feels reasonable though. Another thing to consider is what happens when the particular feature is moved to its final language version. Also, I believe there are many other language specific errors, including the custom ones. And we will be adding them in the future. Does this mean this function requires special maintanense long term?

Reply:
> Does this mean this function requires special maintanense long term?

Yes, I think it does. In particular, the way this manifests is that, when you use an interpolated string conversion, overload resolution can potentially pick a different overload that doesn't use the conversion (even if that conversion wasn't used in the return type, just somewhere in the body). /cc @cston.

Perhaps the thing to do is make this change in dev16.11 (and include the other FeatureIn error codes as well) and do a smoke test on VS to see whether it's a big breaking change?

Reply:
>> Does this mean this function requires special maintanense long term?

> Yes, I think it does. 

I think it will be hard to keep track. We should consider the following:
- Adjusting the code so that it can intersept all current and future language version errors. I think they are already specially recognizable, since the required language version is supplied in a special way fo IDE to recognize.
- Adding a dedicated method in this class that returns true for a language version error and using that helper here. Adding relevant comments and asserts in some strategic places to guide devs into the direction of this new helper.

Reply:
I filed [https://github.com/dotnet/roslyn/issues/53761](https://github.com/dotnet/roslyn/issues/53761) to follow up on this. I'll revert the change in this PR for now, as we discussed offline, and I've added a specific test to cover the difference in behavior.